test/e2e/imports.test.js

```diff
@@ -11,175 +11,137 @@ const sampleMbTilesPath = path.join(
+  'bad-mbtiles/bad-tile-row.mbtiles'
 )
 
-test('GET /imports/:importId returns 404 error when import does not exist', async (t) => {
+test("getImport() returns undefined if the import doesn't exist", async (t) => {
   const server = createServer(t)
 
-  t.is(server.getImport('abc123'), undefined)
+  t.is(server.getImport('abc123'), undefined, 'no import found')
 })
 
-test('GET /imports/progress/:importId returns 404 error when import does not exist', async (t) => {
-  const server = createServer(t).fastifyInstance
-
-  const getImportProgressResponse = await server.inject({
-    method: 'GET',
-    url: `/imports/progress/abc123`,
-  })
+test('getImportProgress() returns an empty iterable if the import does not exist', async (t) => {
+  const server = createServer(t)
 
-  t.equal(getImportProgressResponse.statusCode, 404)
+  const messages = await arrayFrom(server.getImportProgress('abc123'))
+  t.deepEqual(messages, [], 'no messages received')
 })
 
-test('GET /imports/:importId returns import information', async (t) => {
+test('successful import', async (t) => {
   const server = createServer(t)
   const { fastifyInstance } = server
 
   const createImportResponse = await fastifyInstance.inject({
     method: 'POST',
     url: '/tilesets/import',
-    payload: { filePath: sampleSmallMbTilesPath },
+    payload: { filePath: sampleMbTilesPath },
   })
-
   t.equals(createImportResponse.statusCode, 200)
-
   const {
     import: { id: createdImportId },
   } = createImportResponse.json()
 
-  t.ok(server.getImport(createdImportId))
-})
-
-test('GET /imports/progress/:importId returns import progress info (SSE)', async (t) => {
-  const server = createServer(t)
-  const { fastifyInstance } = server
-
-  const createImportResponse = await fastifyInstance.inject({
-    method: 'POST',
-    url: '/tilesets/import',
-    payload: { filePath: sampleMbTilesPath },
-  })
+  const createdImport = server.getImport(createdImportId)
+  t.is(createdImport.state, 'active', 'import is active')
+  t.is(createdImport.error, null, 'import has no errors')
+  t.is(createdImport.finished, null, 'import has not finished')
 
-  const {
-    import: { id: createdImportId },
-  } = createImportResponse.json()
+  const messages = await arrayFrom(server.getImportProgress(createdImportId))
 
-  const address = await fastifyInstance.listen(0, '127.0.0.1')
-  const messages = await importSse(
-    `${address}/imports/progress/${createdImportId}`
-  )
   t.ok(messages.length > 0, 'at least one message is received')
   t.ok(
     messages.every(({ importId }) => importId === createdImportId),
     'all messages have correct importId'
   )
-  const lastMessage = messages[messages.length - 1]
-  t.equal(lastMessage.type, 'complete', 'last message is complete')
-  t.equal(lastMessage.soFar, lastMessage.total)
-
-  t.equal(
-    server.getImport(createdImportId).state,
-    'complete',
-    'import successfully recorded as complete in db'
+  const lastMessage = last(messages)
```

Comment:
more of a meta question, but wondering if it makes sense to avoid using the library helpers (e.g. `arrayFrom()` and `last()`) for tests like this. 

On one hand, I like readable tests and removing boilerplate, but on the other hand, it can be useful to use the "lowest common denominator" approach and not make assumptions about how a module is consumed, although it's slightly more tedious to implement.

Reply:
I went back and forth on this for the tests, but I think you're right. Removed the helpers in c3631bcd2d73aaaf727f2e5d113ffd4379cce222 (and updated our ESLint config to ignore unused variables that start with `_`, just like TypeScript does).